clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```diff
@@ -213,14 +217,16 @@ private void process(final SubscriptionChangeEvent ignored) {
      *              the group is sent out.
      */
     private void process(final UnsubscribeEvent event) {
-        if (!requestManagers.heartbeatRequestManager.isPresent()) {
-            KafkaException error = new KafkaException("Group membership manager not present when processing an unsubscribe event");
-            event.future().completeExceptionally(error);
-            return;
+        if (requestManagers.heartbeatRequestManager.isPresent()) {
+            MembershipManager membershipManager = requestManagers.heartbeatRequestManager.get().membershipManager();
+            CompletableFuture<Void> future = membershipManager.leaveGroup();
+            future.whenComplete(complete(event.future()));
+        } else {
+            // If the group membership manager is not present, we can't send the leave group request,
```

Comment:
nit for a clearer comment maybe: what about just stating that If the consumer is not using the group management capabilities, we still need to clear all assignments it may have? (seems simpler, to the root cause and action needed). The "can't send leave group" is true, but that's just one of the many actions we can't (and actually don't need to) take just because the consumer is not using group mgmt.

Reply:
Update the comment. Thank you.